lib/joins/detect-children-records.js

```diff
@@ -44,7 +44,7 @@ module.exports = function detectChildrenRecords(primaryKeyAttr, records) {
 
     // Process each key in the record and build up a child record that can
     // then be nested into the parent.
-    _.each(record, function checkForChildren(val, key) {
+    _.forIn(record, function checkForChildren(val, key) {
```

Comment:
For posterity: 
![image](https://user-images.githubusercontent.com/618009/39908376-7a0baa4c-54b3-11e8-8803-69b4a1547649.png)